saleor/graphql/checkout/mutations.py

```diff
@@ -1035,6 +1053,178 @@ def perform_mutation(
+            "Updates the delivery method (shipping method or pick up point)"
+            "of the checkout."
+        )
+        error_type_class = CheckoutError
+        error_type_field = "checkout_error"
+
+    @classmethod
+    def perform_on_shipping_method(
+        cls, info, shipping_method_id, checkout_info, lines, checkout, manager
+    ):
+        shipping_method = cls.get_node_or_error(
+            info,
+            shipping_method_id,
+            only_type=ShippingMethod,
+            field="shipping_method_id",
+            qs=shipping_models.ShippingMethod.objects.prefetch_related(
+                "postal_code_rules"
+            ),
+        )
+
+        CheckoutDeliveryMethodUpdate._check_delivery_method(
+            checkout_info, lines, shipping_method=shipping_method, collection_point=None
+        )
+
+        checkout_delivery_method_update = (
+            CheckoutDeliveryMethodUpdate._update_delivery_method(
+                manager,
+                checkout,
+                shipping_method=shipping_method,
+                collection_point=None,
+            )
+        )
+        recalculate_checkout_discount(
+            manager, checkout_info, lines, info.context.discounts
+        )
+        return checkout_delivery_method_update
+
+    @classmethod
+    def perform_on_collection_point(
+        cls, info, collection_point_id, checkout_info, lines, checkout, manager
+    ):
+        collection_point = cls.get_node_or_error(
+            info,
+            collection_point_id,
+            only_type=Warehouse,
+            field="collection_point_id",
+            qs=warehouse_models.Warehouse.objects.select_related("address"),
+        )
+        CheckoutDeliveryMethodUpdate._check_delivery_method(
+            checkout_info,
+            lines,
+            shipping_method=None,
+            collection_point=collection_point,
+        )
+        return CheckoutDeliveryMethodUpdate._update_delivery_method(
+            manager, checkout, shipping_method=None, collection_point=collection_point
+        )
+
+    @staticmethod
+    def _check_delivery_method(
+        checkout_info,
+        lines,
+        *,
+        shipping_method: Optional[ShippingMethod],
+        collection_point: Optional[Warehouse]
+    ) -> None:
+        delivery_method = shipping_method
+        error_msg = "This shipping method is not applicable."
+
+        if collection_point is not None:
+            delivery_method = collection_point
+            error_msg = "This pick up point is not applicable."
+
+        delivery_method_is_valid = clean_delivery_method(
+            checkout_info=checkout_info, lines=lines, method=delivery_method
+        )
+        if not delivery_method_is_valid:
+            raise ValidationError(
+                {
+                    "delivery_method": ValidationError(
+                        error_msg,
+                        code=CheckoutErrorCode.SHIPPING_METHOD_NOT_APPLICABLE.value,
```

Comment:
Is it ok, that for both (shipping_method and c_andC) we have the same error code? The name of this suggest that error code is related to the shipping method.

Reply:
I have an add additional Error code, done.